### DISCUSSION TITLE
Where is the "next start" Js File ? how can i run with node command ?

### DISCUSSION THREAD

**Discussion by sromexs** [vercel/next.js#14145] [Help] [answered]
hi
my web hosting for start node server just accepts a `app.js` file path to run my app.
i wondering how can i start my app with like `node next.js` instead of `npm start` ?
is there any file i can start that file with node ?

**Comment by illuminist** [comment, marked as answer]
First choice is to create a server instance in `app.js`, by using [custom server](https://nextjs.org/docs/advanced-features/custom-server)

```js
const { createServer } = require('http')
const { parse } = require('url')
const next = require('next')

const dev = process.env.NODE_ENV !== 'production'
const app = next({ dev })
const handle = app.getRequestHandler()

app.prepare().then(() => {
  createServer((req, res) => {
    // Be sure to pass `true` as the second argument to `url.parse`.
    // This tells it to parse the query portion of the URL.
    const parsedUrl = parse(req.url, true)
    handle(req, res, parsedUrl)
  }).listen(3000, (err) => {
    if (err) throw err
    console.log('> Ready on http://localhost:3000')
  })
})
```

Second choice is to spawn child process
```js
const {exec} = require('child_process');
exec('npm start')
```

**Reply by sromexs** [reply]
Thanks First One worked
But child process not working
can you work on child process ?
it think that is better solution

**Reply by illuminist** [reply]
Not sure if your host allows to spawn second node process, so the first method should be safer.
